### PR TITLE
[AppService] `az functionapp create`: Use app insights connection string instead of instrumentation key

### LINKED ISSUE #27607: az functionapp create does not use newly created application insight's connection string
### Describe the bug

`az functionapp create` use `APPINSIGHTS_INSTRUMENTATIONKEY` instead of `APPLICATIONINSIGHTS_CONNECTION_STRING `.

<App settings reference for Azure Functions | Microsoft Learn>
https://learn.microsoft.com/en-us/azure/azure-functions/functions-app-settings#appinsights_instrumentationkey
> While the use of APPLICATIONINSIGHTS_CONNECTION_STRING is recommended in all cases

> 

### Related command

```
az functionapp create \
          --resource-group ${RESOURCE_GROUP_NAME} \
          --storage-account ${STORAGE_NAME}\
          --plan ${ASP_NAME} \
          --name  ${FUINCTIONAPP_NAME}  \
          --functions-version 4 \
          --os-type Linux \
          --runtime custom \
           --verbose --debug
```

### Errors

The command will not fail.

### Issue script & Debug output

While processing command newly Application Insights component will be created. I expect Function App to have app settings with `APPLICATIONINSIGHTS_CONNECTION_STRING`  with the value of ConnectionString from Application insights, but app settings has  `APPINSIGHTS_INSTRUMENTATIONKEY`.

### Expected behavior

Newly created FunctionApp should have app settings with `APPLICATIONINSIGHTS_CONNECTION_STRING`.

### Environment Summary

azure-cli                         2.53.0

core                              2.53.0
telemetry                          1.1.0

Extensions:
application-insights               1.0.0
communication                    1.7.2b1
log-analytics                      0.2.2

Dependencies:
msal                            1.24.0b2
azure-mgmt-resource             23.1.0b2

Python location '/opt/az/bin/python3'
Extensions directory '/home/toshida/.azure/cliextensions'

Python (Linux) 3.10.10 (main, Sep 20 2023, 06:07:38) [GCC 11.4.0]

Legal docs and information: aka.ms/AzureCliLegal

Your CLI is up-to-date.

### Additional context

I think below block is related.
https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L4059-L4064

### PR DESCRIPTION
**Pull request by kamperiadis**
**Related command**
`az functionapp create`

**Description**
Resolves #27607 

**Testing Guide**
Create function app and check that app setting added is `APPLICATIONINSIGHTS_CONNECTION_STRING` instead of `APPINSIGHTS_INSTRUMENTATION_KEY`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

**Comment by zhoxing-ms** [comment]
@kamperiadis Will the behavioral changes here cause breaking changes to the automation scripts of customers?

**Comment by kamperiadis** [comment]
@zhoxing-ms This is not a breaking change as using either app setting would work. These changes are made for [this effort](https://learn.microsoft.com/en-us/azure/azure-monitor/app/migrate-from-instrumentation-keys-to-connection-strings) to migrate from the instrumentation key to the connection string.

**Comment by zhoxing-ms** [comment]
> This is not a breaking change as using either app setting would work. These changes are made for [this effort](https://learn.microsoft.com/en-us/azure/azure-monitor/app/migrate-from-instrumentation-keys-to-connection-strings) to migrate from the instrumentation key to the connection string.

Okay, so this is just an internal logic modification and won't have any impact on the customers' script, right?

**Comment by zhoxing-ms** [comment]
Could you please resolve these CI issues?

**Comment by kamperiadis** [comment]
> > This is not a breaking change as using either app setting would work. These changes are made for [this effort](https://learn.microsoft.com/en-us/azure/azure-monitor/app/migrate-from-instrumentation-keys-to-connection-strings) to migrate from the instrumentation key to the connection string.
> 
> Okay, so this is just an internal logic modification and won't have any impact on the customers' script, right?

That is correct.

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py [hunk 1]
```diff
@@ -1191,6 +1191,26 @@ def test_functionapp_with_app_insights_key(self, resource_group, storage_account
             'functionapp delete -g {} -n {}'.format(resource_group, functionapp_name))
 
 
+class FunctionAppWithAppInsightsConnString(ScenarioTest):
+    def __init__(self, method_name, config_file=None, recording_name=None, recording_processors=None, replay_processors=None, recording_patches=None, replay_patches=None, random_config_dir=False):
+        super().__init__(method_name, config_file, recording_name, recording_processors, replay_processors, recording_patches, replay_patches, random_config_dir)
+        self.cmd('extension add -n application-insights')
```
Comment (bebound) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py:1197]: `extension add -n` should not be here. It causes CI failure:
```
==================================== ERRORS ====================================
__________ ERROR collecting tests/latest/test_functionapp_commands.py __________
/usr/lib64/az/lib/python3.9/site-packages/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py:990: in __init__
    self.cmd('extension add -n application-insights')
/usr/lib64/az/lib/python3.9/site-packages/azure/cli/testsdk/base.py:176: in cmd
    return execute(self.cli_ctx, command, expect_failure=expect_failure).assert_with_checks(checks)
/usr/lib64/az/lib/python3.9/site-packages/azure/cli/testsdk/base.py:260: in __init__
    raise AssertionError('The command failed. Exit code: {}'.format(self.exit_code))
E   AssertionError: The command failed. Exit code: 1
__________ ERROR collecting tests/latest/test_functionapp_commands.py __________
/usr/lib64/az/lib/python3.9/site-packages/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py:1579: in __init__
    self.cmd('extension add -n application-insights')
/usr/lib64/az/lib/python3.9/site-packages/azure/cli/testsdk/base.py:176: in cmd
    return execute(self.cli_ctx, command, expect_failure=expect_failure).assert_with_checks(checks)
/usr/lib64/az/lib/python3.9/site-packages/azure/cli/testsdk/base.py:260: in __init__
    raise AssertionError('The command failed. Exit code: {}'.format(self.exit_code))
E   AssertionError: The command failed. Exit code: 1
__________ ERROR collecting tests/latest/test_functionapp_commands.py __________
/usr/lib64/az/lib/python3.9/site-packages/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py:1707: in __init__
    self.cmd('extension add -n application-insights')
/usr/lib64/az/lib/python3.9/site-packages/azure/cli/testsdk/base.py:176: in cmd
    return execute(self.cli_ctx, command, expect_failure=expect_failure).assert_with_checks(checks)
/usr/lib64/az/lib/python3.9/site-packages/azure/cli/testsdk/base.py:260: in __init__
    raise AssertionError('The command failed. Exit code: {}'.format(self.exit_code))
E   AssertionError: The command failed. Exit code: 1
---------- generated xml file: /azure_cli_test_result/appservice.xml -----------
=========================== short test summary info ============================
ERROR tests/latest/test_functionapp_commands.py::FunctionAppManagedEnvironment
ERROR tests/latest/test_functionapp_commands.py::FunctionAppWithAppInsightsConnString
ERROR tests/latest/test_functionapp_commands.py::FunctionappAppInsightsWorkspace

```

https://dev.azure.com/azclitools/public/_build/results?buildId=209680&view=logs&j=fce0fe10-dd1b-5814-f04b-ea25f95c49f0&t=f9fbf0f4-0b46-5841-08eb-fc21c02e7e0a&l=3124

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py:1197]: @kamperiadis Could you please take a look at this issue?

Comment (kamperiadis) [reply src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py:1197]: @zhoxing-ms I've created this PR to solve this issue: https://github.com/Azure/azure-cli/pull/30636

**Pull request merged**